sensor/admission-control/manager/manager_impl.go

```diff
@@ -181,20 +204,23 @@ func (m *manager) runSettingsWatch() {
 			return
 		case newSettings := <-m.settingsC:
 			m.ProcessNewSettings(newSettings)
-		}
-	}
-}
-
-func (m *manager) runUpdateResourceReqWatch() {
-	defer m.stoppedSig.Signal()
-	defer log.Info("Stopping watcher for new sensor events")
-
-	for {
-		select {
-		case <-m.stopSig.Done():
-			return
 		case req := <-m.resourceUpdatesC:
 			m.processUpdateResourceRequest(req)
+		default:
+			// Select on syncC only if there is nothing to be read from the main
+			// channels. The duplication of select branches is a bit ugly, but inevitable
+			// without reflection.
+			select {
+			case <-m.stopSig.Done():
+				m.ProcessNewSettings(nil)
+				return
+			case newSettings := <-m.settingsC:
+				m.ProcessNewSettings(newSettings)
+			case req := <-m.resourceUpdatesC:
+				m.processUpdateResourceRequest(req)
+			case sig := <-m.syncC:
```

Comment:
With `stopSig` and `stoppedSig` already being mentioned in this function, perhaps call this `syncSig` to be a bit more specific?

Reply:
done